### PR TITLE
Query and update

### PR DESCRIPTION
**Pull request by alexmedinasf**
In this project, I used transactions to update and delete records. I also used my knowledge of aggregate functions and GROUP BY to answer analytical questions.

[LINK TO THE SCREENSHOTS](https://imgur.com/a/NUtlriw) hosted on Imgur. PS: ITS SAYING THAT MAY CONTAIN 18+ IMAGES BUT ITS JUST THE SCREENSHOTS OF THE PGADMIN4 RESULTS

edit: changed the link with the updated photos.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -13,4 +13,33 @@ INSERT INTO public.animals(
...
+	name, date_of_birth, escape_attempts, neutered, weight_kg)
+	VALUES ('Plantmon', '2021-11-15', '2', '1', '5.7');
+
+	INSERT INTO public.animals(
+	name, date_of_birth, escape_attempts, neutered, weight_kg)
+	VALUES ('Squirtle', '1993-04-02', '3', '0', '12.13');
+
+	INSERT INTO public.animals(
+	name, date_of_birth, escape_attempts, neutered, weight_kg)
+	VALUES ('Angemon', '2005-06-12', '1', '1', '45');
```
Comment (AbelG101) [inline data.sql]: - [x] Please make sure to assign the proper weight for the first 4 animals as specified in the project requirements as they should have a negative prefix.

File: queries.sql
Comment (AbelG101) [inline queries.sql:1]: - [x] It seems like you're forgetting the query for this question "Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction."  from the project requirements so please add it accordingly.

File: queries.sql [hunk 2]
```diff
@@ -21,3 +21,74 @@ WHERE name != 'Gabumon';
...
+
+update animals
+set species = 'digimon'
+where name like '%mon';
+
+update animals
+set species = 'pokemon'
+where species is null;
+
+-- deleting animals and rollback
```
Comment (AbelG101) [inline queries.sql]: - [x] Please make sure to add the above SQL statements in a transaction with proper `BEGIN`, `COMMIT` and `ROLLBACK` and also make sure to verify that the change was made before rolling back and after making a commit with `SELECT` statements.

File: queries.sql [hunk 3]
```diff
@@ -21,3 +21,74 @@ WHERE name != 'Gabumon';
...
+
+begin;
+
+delete from animals;
+
+rollback;
```
Comment (AbelG101) [inline queries.sql]: - [x] Please make sure to add a statement to verify if all records in the animals table still exist after rolling back the deleted animals table as specified in the project requirements.

File: queries.sql [hunk 4]
```diff
@@ -21,3 +21,74 @@ WHERE name != 'Gabumon';
...
+update animals
+set weight_kg = weight_kg * -1;
+
+rollback to savepoins_1;
+
+-- updating animals weight -- 
+
+update animals
+set weight_kg = weight_kg * -1
+where weight_kg < 1;
```
Comment (AbelG101) [inline queries.sql]: - [x] As per the project requirements you should update all animals' weights that are negative so please update the statement at line `61` to `where weight_kg < 0;`

File: queries.sql [hunk 5]
```diff
@@ -21,3 +21,74 @@ WHERE name != 'Gabumon';
...
+
+--What is the average weight of animals?
+
+select avg(weight_kg) from animals;
+
+--Who escapes the most, neutered or not neutered animals? --
+
+select count(escape_attempts) from animals
+group by neutered;
+
```
Comment (AbelG101) [inline queries.sql]: - [x] Ouch! this query doesn't necessarily show whether it's the "neutered" or "non-neutered" that escapes the most you might need to add the `neutered` column or another adjustment.

File: queries.sql
Comment (AbelG101) [inline queries.sql:1]: - [x] And finally, make sure to update your screenshots with respect to the changes mentioned here.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @alexmedinasf 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Species column is properly added ✔
- Proper transaction management ✔
- Correct use of Git Flow ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @alexmedinasf 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Species column is properly added ✔
- Correct queries for each question ✔
- Proper transaction management ✔
- Very well-organized SQL statements✔
- Correct use of Git Flow ✔
- Descriptive PR title with a great screenshot summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alexmedinasf** [comment]
Thank you for the review!

**Pull request merged**